src/database/UIDatabase.js

```diff
@@ -133,6 +133,29 @@ class UIDatabase {
     return true;
   }
 
+  /**
+   * Imports the realm file from the location chosen in the open dialog box.
+   * May request storage permissions if required.
+   * @returns Returns `True` if data is imported successfully. Otherwise `False`
+   */
+  async importData() {
+    const { realm } = this.database;
+    const { path: realmPath } = realm;
+    const realmFilePath = await selectDocument({ fileType: 'realm' });
+
+    if (realmFilePath === '') return false;
+    if (realm.isInTransaction) return false;
+
+    // Proceed importation of the realm file to the app database location
+    try {
+      await RNFS.copyFile(realmFilePath, realmPath);
```

Comment:
after this copy is finished, is it possible to read the database version? ( sorry, don't have time to check right now )
It would be awesome if the import could check, alert if the version is invalid, and then rollback the change.

Then you won't need the version check and alert - I find that really tiresome in mSupply desktop. Once you load up a newer datafile, you can't easily get back. It would be good if the import checked and didn't allow screwing up the app.

Reply:
Agree. I had also an initial thought about as a preliminary actions for the import functionality but I found a bit difficult to read database version from the external realm file from the app (not sure if there's an easy way to do it) so that version incompatibility can be checked before importation. Also the current mobile app doesn't keep track of app's version in anywhere. So those things made me to decide checking on the startup of the app.

Reply:
On the second thought, here are changes 8b7f50ed @mark-prins. Now the version incompatibility is checked before importing datafile not on the startup of the app.

Reply:
That's great! thanks :)